Added TCP client docs from v5.0.9 to 20.10 and 21.2

oskardudycz opened the pull request:
Copy/pasted TCP client docs from v5.0.9 to 20.10 and 21.2.
Updated compatibility mode docs for the 20.10 and 21.2 version.

Fixes #282 
Fixes #283

oskardudycz committed:
Added TCP client docs for 20.10 and 21.2

oskardudycz committed:
Updated TCP code samples to match the expected client NuGet version

oskardudycz committed:
Updated the compatibility mode docs for v20.10 and v21.2

oskardudycz committed:
Updated TCP client samples links
Updated linking in cloud to target 20.10 instead of v5
Updated the 21.2 compatibility mode description

hayley-jean commented on docs/clients/dotnet/20.10/connecting/compatibility-mode.md:
```diff
@@ -0,0 +1,18 @@
+# Compatibility Mode
+
+Compatibility Mode was added in v5.0.10 and v21.2.0 of the client to set certain configuration options depending on the Event Store version the client is connecting to. It allowa discovering whether the Event Store server is a v5 or v20 server.
```
```suggestion
Compatibility Mode was added in v5.0.10 and v21.2.0 of the client to set certain configuration options depending on the Event Store version the client is connecting to. It allows discovering whether the Event Store server is a v5 or v20 server.
```

oskardudycz replied:
Fixed in https://github.com/EventStore/documentation/pull/288/commits/151543c1acbb559165d3d983279156be408f5eb2.

hayley-jean commented on docs/clients/dotnet/20.10/embedded/README.md:
```diff
@@ -0,0 +1,192 @@
+# Embedded client
```
We haven't released the embedded client for v20.10 and above, should we remove the docs relating to it?

oskardudycz replied:
Nice catch - removed in https://github.com/EventStore/documentation/pull/288/commits/ba3f7e7c27574e7c876a5ae749868472ce656f95.

hayley-jean commented on docs/clients/dotnet/21.2/connecting/compatibility-mode.md:
```diff
@@ -0,0 +1,66 @@
+
+At the moment, the v21.2 client supports two modes:
+
+- "5" to allow connection to Event Store servers v5.0.10.
```
```suggestion
- "5" to allow connection to v5 Event Store servers.
```
Auto-compatibility is compatible with any version of v5

oskardudycz replied:
Fixed in https://github.com/EventStore/documentation/pull/288/commits/151543c1acbb559165d3d983279156be408f5eb2.

hayley-jean commented on docs/clients/dotnet/21.2/connecting/compatibility-mode.md:
```diff
@@ -0,0 +1,66 @@
+
+## v5 Compatibility Mode
+
+The v5 Compatibility Mode allows the v21.2 client to connect to a cluster running in ES Cloud.
```
```suggestion
The v5 Compatibility Mode allows the v21.2 client to connect to a v5 cluster.
```

oskardudycz replied:
Fixed in https://github.com/EventStore/documentation/pull/288/commits/151543c1acbb559165d3d983279156be408f5eb2.

hayley-jean commented on docs/clients/dotnet/21.2/connecting/compatibility-mode.md:
````diff
@@ -0,0 +1,66 @@
+For example:
+
+```csharp
+var connectionString = $"ConnectTo=discover://{CloudClusterId}.mesdb.eventstore.cloud:2113?UseSslConnection=true;ValidateServer=true;CompatibilityMode=5;"
````
```suggestion
var connectionString = $"ConnectTo=discover://{cluster_dns}:2113?TargetHost={cluster_dns};CompatibilityMode=5;"
```

oskardudycz replied:
Fixed in https://github.com/EventStore/documentation/pull/288/commits/151543c1acbb559165d3d983279156be408f5eb2.

hayley-jean commented on docs/clients/dotnet/21.2/embedded/README.md:
```diff
@@ -0,0 +1,192 @@
+# Embedded client
```
See previous comment about embedded client

oskardudycz replied:
Nice catch - removed in https://github.com/EventStore/documentation/pull/288/commits/ba3f7e7c27574e7c876a5ae749868472ce656f95.

oskardudycz committed:
Removed embedded client from 20.10 and 21.2 docs (It's not available in those versions)

oskardudycz committed:
Updated mishaps in compatibility mode docs for v20.10 and v21.2

oskardudycz committed:
Added connection string param description for the Compatibility mode
Renamed Connection option docs to connection settings
Added documentation for TCP client Compatibility Mode setting

oskardudycz committed:
Added redirect from TCP Client Options to renamed TCP Client Settings to keep the backward compatibility
